azure-devops/azext_devops/dev/repos/policy.py

```diff
@@ -0,0 +1,371 @@
+                                           REQUIRED_REVIEWER_POLICY,
+                                           REQUIRED_REVIEWER_POLICY_ID)
+from azext_devops.dev.common.identities import resolve_identity_as_id
+
+logger = get_logger(__name__)
+
+
+def list_policy(organization=None, project=None, detect=None):
+    """
+    :param organization: Azure Devops organization URL. Example: https://dev.azure.com/MyOrganizationName/
+    :type organization: str
+    :param project: Name or ID of the project.
+    :type project: str
+    :param detect: Automatically detect organization. Default is "on".
+    :type detect: str
+    :rtype: [PolicyConfiguration]
+    """
+    try:
+        organization, project = resolve_instance_and_project(
+            detect=detect, organization=organization, project=project)
+        policy_client = get_policy_client(organization)
+        return policy_client.get_policy_configurations(project=project)
+    except VstsServiceError as ex:
+        raise CLIError(ex)
+
+
+def get_policy(id, organization=None, project=None, detect=None):  # pylint: disable=redefined-builtin
+    """
+    :param id: ID of the policy.
+    :type id: int
+    :param organization: Azure Devops organization URL. Example: https://dev.azure.com/MyOrganizationName/
+    :type organization: str
+    :param project: Name or ID of the project.
+    :type project: str
+    :param detect: Automatically detect organization. Default is "on".
+    :type detect: str
+    :rtype: :class:`<PolicyConfiguration> <policy.v4_0.models.PolicyConfiguration>`
+    """
+    try:
+        organization, project = resolve_instance_and_project(
+            detect=detect, organization=organization, project=project)
+        policy_client = get_policy_client(organization)
+        return policy_client.get_policy_configuration(project=project, configuration_id=id)
+    except VstsServiceError as ex:
+        raise CLIError(ex)
+
+
+def delete_policy(id, organization=None, project=None, detect=None):  # pylint: disable=redefined-builtin
+    """
+    :param id: ID of the policy.
+    :type id: int
+    :param organization: Azure Devops organization URL. Example: https://dev.azure.com/MyOrganizationName/
+    :type organization: str
+    :param project: Name or ID of the project.
+    :type project: str
+    :param detect: Automatically detect organization. Default is "on".
+    :type detect: str
+    """
+    try:
+        organization, project = resolve_instance_and_project(
+            detect=detect, organization=organization, project=project)
+        policy_client = get_policy_client(organization)
+        return policy_client.delete_policy_configuration(project=project, configuration_id=id)
+    except VstsServiceError as ex:
+        raise CLIError(ex)
+
+# pylint: disable=too-many-locals
+def create_policy(repository_id, branch,
+                  isBlocking=False, isEnabled=False,
+                  policy_type=None,
```

Comment:
Is policy_type also a required param?

Reply:
no, this is not required.
This can be passed as policy_configuration

Reply:
What if both policy_type and policy_configuration are None

Reply:
we will throw exception